.github/actions/install-risc0/action.yml

```diff
@@ -2,11 +2,20 @@ name: "Install Risc0"
 
 description: "Install the Risc0 toolchain. Requires the Rust toolchain to be installed."
 
+inputs:
+  github_token:
+    description: "The GitHub token to use for rate limiting."
+    required: true
+
 runs:
   using: "composite"
   steps:
     - name: Install cargo-binstall
       uses: cargo-bins/cargo-binstall@v1.11.1
+      env:
+        GITHUB_TOKEN: ${{ inputs.github_token }}
     - name: Install risc0 using cargo-binstall
       run: cargo binstall -y cargo-risczero@1.2.4 && cargo risczero install
```

Comment:
Seems that node with risc0`1.2.4` fails more often, lets try `1.2.0`?

Reply:
Had to do a bit of fire extinguishing due to some deps now requiring rust version 2024, meaning we had to bump the version in the CI, which I though was done automatically via the toolchain file, but it seems not to be the case. Bumping to 1.85 also introduced some more clippy warnings, which I had to address.

Reply:
Nevertheless, the tests seem to be failing also with this version installed.

Reply:
Should I try a different version? Do you have any hints?

Reply:
I'm working on removing `indexer_tests`, the most important for now is that `two_nodes_happy` tests works reliably (it tests consensus). I see that the situation changed, as we have macos green. It seems that now it fails less often - let's try updating to `1.2.4` once I remove `indexer_test`?